build_scripts/tokenize.py

```diff
@@ -0,0 +1,331 @@
+    root_length, child_length = len(root_line), len(child_line)
+
+    # Retrieve other meta info for lines
+    meta_root = {
+        "commit_hash": commit_list[0]["Commit_Hash"],
+        "author_name": parse_author(commit_list[0]["Author"])[0],
+        "email": parse_author(commit_list[0]["Author"])[1],
+        "time": format_date(commit_list[0]["Date"])
+    }
+
+    meta_child = {
+        "commit_hash": commit_list[1]["Commit_Hash"],
+        "author_name": parse_author(commit_list[1]["Author"])[0],
+        "email": parse_author(commit_list[1]["Author"])[1],
+        "time": format_date(commit_list[1]["Date"])
+    }
+
+    # Return val
+    segments = []
+
+    # Find overlap at the start
+    start_idx = 0
+    for i in range(min(root_length, child_length)):
+        if root_line[i] != child_line[i]:
+            break
+        start_idx = i + 1
+
+    # Find overlap at the end
+    end_idx = 0
+    for i in range(1, min(root_length, child_length) + 1):
+        if root_line[-i] != child_line[-i]:
+            break
+        end_idx = i
+
+    start_overlap = root_line[:start_idx]
+    end_overlap = root_line[-end_idx:] if end_idx > 0 else ""
+    mid_segment = root_line[start_idx:root_length -
+                            end_idx] if start_idx < root_length - end_idx else ""
+
+    # Case 1: start + mid + end
+    if start_overlap and end_overlap and mid_segment:
+        segments.append({"token": start_overlap, "meta": meta_child})
+        segments.append({"token": mid_segment, "meta": meta_root})
+        segments.append({"token": end_overlap, "meta": meta_child})
+
+    # Case 2: start + end (mid is empty)
+    elif start_overlap and not mid_segment and end_overlap:
+        segments.append({"token": start_overlap, "meta": meta_child})
+        segments.append({"token": end_overlap, "meta": meta_child})
+
+    # Case 3: mid + end (start is empty)
+    elif not start_overlap and mid_segment and end_overlap:
+        segments.append({"token": mid_segment, "meta": meta_root})
+        segments.append({"token": end_overlap, "meta": meta_child})
+    # Case 4: no overlap
+    else:
+        segments.append({"token": root_line, "meta": meta_root})
+
+    return segments
+
+
+# Main execution
+if __name__ == '__main__':
+    # Parse arguments
+    parser = argparse.ArgumentParser(description='Process some files.')
+    parser.add_argument('git_diff_report', type=str,
+                        help='Path to git_diff_report.txt')
+    parser.add_argument('intermediate_file', type=str,
+                        help='Path to intermediate_file')
+    parser.add_argument('output_file', type=str, help='Path to output_file')
+    parser.add_argument('tag1', type=str, help='old version tag')
+    parser.add_argument('tag2', type=str, help='new verison tag')
+
+    args = parser.parse_args()
+
+    git_diff_report_path = args.git_diff_report
+    intermediate_file_path = args.intermediate_file
```

Comment:
Is this intermediate_file something that the user would ever want or use? 

If not,

- Are we cleaning it up? (I might just have missed it)
- Suggest removing it from the argument list and just creating a random file name. There nice ways to do this in Python as I recall, Google is your friend.

Reply:
- We are not cleaning it up. But modified the file to clean it up in the new commit. 
- Have just removed it from the arg list, let py create a random file name, and removed it right before the program exits.